crates/compiler/src/parser.rs

```diff
@@ -493,6 +493,8 @@ where
     fn parse_text(&mut self, text: VStr<'a>) {
         let mut text = smallvec![text];
         let mut next_token = None;
+        let start = self.tokens.last_position();
+        let location = self.tokens.get_location_from(start);
```

Comment:
I don't think this is a correct fix. Note the location only contains the text span.